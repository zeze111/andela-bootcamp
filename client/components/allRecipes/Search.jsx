import React from 'react';
import PropTypes from 'prop-types';

const Search = props => (
  <form className="col s7 offset-s4">
    <div className="row">
      <div className="input-field col s6">
        <label htmlFor="search" />
        <input
          placeholder="Search for Recipes"
          type="text"
          id="search"
          value={props.search}
          onChange={props.onChange}
        />
      </div>
      <div className="col s6">
        <button
          className="btn grey top"
          type="button"
        >
          <i className="material-icons">search</i>
        </button>
      </div>
    </div>
  </form>
);

Search.propTypes = {
  search: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Search;
