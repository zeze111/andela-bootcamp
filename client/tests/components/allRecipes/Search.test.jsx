import React from 'react';

import Search from '../../../components/allRecipes/Search';
import store from '../../../store';

const props = {
  onChange: jest.fn(() => Promise.resolve()),
  search: 'some'
};

describe('Test for Search', () => {
  it('should render component', () => {
    const wrapper = shallow(<Search {...props} store={store}/> );
    expect(wrapper).toMatchSnapshot()
  });
});
