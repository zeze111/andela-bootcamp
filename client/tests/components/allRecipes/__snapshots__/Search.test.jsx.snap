// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Search should render component 1`] = `
<form
  className="col s12 offset-s1 offset-m3 offset-l5 paginate"
>
  <div
    className="row"
  >
    <div
      className="input-field col l6 m9 s12"
    >
      <i
        className="material-icons search-icon"
      >
        search
      </i>
      <input
        className="search-input"
        id="search"
        onChange={[MockFunction]}
        placeholder="Search for recipes"
        type="text"
        value="some"
      />
    </div>
  </div>
</form>
`;
